core/src/main/java/org/apache/iceberg/DefaultValueParser.java

```diff
@@ -0,0 +1,374 @@
+            defaultValue instanceof Boolean, "Invalid default %s value: %s", type, defaultValue);
+        generator.writeBoolean((Boolean) defaultValue);
+        break;
+      case INTEGER:
+        Preconditions.checkArgument(
+            defaultValue instanceof Integer, "Invalid default %s value: %s", type, defaultValue);
+        generator.writeNumber((Integer) defaultValue);
+        break;
+      case LONG:
+        Preconditions.checkArgument(defaultValue instanceof Long, "Invalid default %s value: %s", type, defaultValue);
+        generator.writeNumber((Long) defaultValue);
+        break;
+      case FLOAT:
+        Preconditions.checkArgument(defaultValue instanceof Float, "Invalid default %s value: %s", type, defaultValue);
+        generator.writeNumber((Float) defaultValue);
+        break;
+      case DOUBLE:
+        Preconditions.checkArgument(defaultValue instanceof Double, "Invalid default %s value: %s", type, defaultValue);
+        generator.writeNumber((Double) defaultValue);
+        break;
+      case DATE:
+        Preconditions.checkArgument(
+            defaultValue instanceof Integer, "Invalid default %s value: %s", type, defaultValue);
+        generator.writeString(DateTimeUtil.daysToIsoDate((Integer) defaultValue));
+        break;
+      case TIME:
+        Preconditions.checkArgument(
+            defaultValue instanceof Long, "Invalid default %s value: %s", type, defaultValue);
+        generator.writeString(DateTimeUtil.microsToIsoTime((Long) defaultValue));
+        break;
+      case TIMESTAMP:
+        Preconditions.checkArgument(
+            defaultValue instanceof Long, "Invalid default %s value: %s", type, defaultValue);
+        if (((Types.TimestampType) type).shouldAdjustToUTC()) {
+          generator.writeString(DateTimeUtil.microsToIsoDateTimeTz((Long) defaultValue));
+        } else {
+          generator.writeString(DateTimeUtil.microsToIsoDateTime((Long) defaultValue));
+        }
+        break;
+      case STRING:
+        Preconditions.checkArgument(
+            defaultValue instanceof CharSequence, "Invalid default %s value: %s", type, defaultValue);
+        generator.writeString(((CharSequence) defaultValue).toString());
+        break;
+      case UUID:
+        Preconditions.checkArgument(defaultValue instanceof UUID, "Invalid default %s value: %s", type, defaultValue);
+        generator.writeString(defaultValue.toString());
+        break;
+      case FIXED:
+        Preconditions.checkArgument(
+            defaultValue instanceof ByteBuffer, "Invalid default %s value: %s", type, defaultValue);
+        ByteBuffer byteBufferDefaultValue = (ByteBuffer) defaultValue;
+        Preconditions.checkArgument(
+            byteBufferDefaultValue.hasArray() && byteBufferDefaultValue.arrayOffset() == 0 &&
+                byteBufferDefaultValue.position() == 0,
+            "Invalid default %s value: %s, not a valid bytebuffer representation",
+            type, defaultValue);
```

Comment:
This is incorrect. Any `ByteBuffer` is a valid value that is supported by `ByteBuffers.toByteArray`.

I think the confusion comes from [this comment](https://github.com/apache/iceberg/pull/4871/files#discussion_r908731811). The check I intended is `((ByteBuffer) defaultValue).remaining() == expectedLength`.

Reply:
fixed